air-quality-backend/src/air_quality/database/forecasts.py

```diff
@@ -22,3 +23,24 @@ def delete_data_before(forecast_valid_time: datetime):
         {"forecast_valid_time": {"$lt": forecast_valid_time}}
     )
     logging.info(f"Deleted {result.deleted_count} documents from forecast_data")
+
+
+def get_forecast_data_from_database(
+    valid_date_from: datetime,
+    valid_date_to: datetime,
+    location_type: str,
+    forecast_base_time: datetime,
+    location_name: str,
+):
+    collection = get_collection("forecast_data")
+    query = {
+        "location_type": location_type,
+        "forecast_base_time": forecast_base_time,
+        "forecast_valid_time": {
+            "$gte": valid_date_from,
+            "$lt": valid_date_to,
+        },
+    }
+    if location_name is not None:
+        query["name"] = location_name
+    return map_forecast(list(collection.find(query)))
```

Comment:
Ah I missed this earlier - I think we should call map_forecast inside the API part of the code rather than in database. That way we separate the api concern from the database and we remove a circular dependency